lib/openstack_query/docs/PROPERTIES.md

```diff
@@ -0,0 +1,117 @@
+# Query Properties
+
+Properties refer to the different key-word arguments that are stored in the Openstack resource.
+
+Properties for each query are defined by Property Enums
+- This is done so that names of each property are less ambiguous.
+- Allows us to implement aliases for these properties that are independent of Openstacksdk
+
+For each Query object, there is a corresponding enum class `<Resource>Properties` that is used for storing properties.
+- located in `lib/enums/query/props` - usually in file called `<resource>_properties.py`
+
+
+# Property Class Reference
+
+| Property Class      | Description                                                                      |
+|---------------------|----------------------------------------------------------------------------------|
+| `ServerProperties`  | Properties related to Openstack Servers (VMs). Used by `ServerQuery` Query class |
+| `UserProperties`    | Properties related to Openstack Users. Used by `UserQuery` Query class           |
+| `ProjectProperties` | Properties related to Openstack Projects. Used by `ProjectQuery` Query class     |
+| `FlavorProperties`  | Properties related to Openstack Flavors. Used by `FlavorQuery` Query class       |
+
+
+## Property Reference
+
+`ServerProperties`
```

Comment:
IMO each of these should be in their own files:

```
lib/openstack_query/docs/properties/server_properties.md
lib/openstack_query/docs/properties/user_properties.md
```

That way I can easily get to the relevant docs in vim/emacs/code/GH based on a file name rather than open a massive file and have to jump around

It also makes it trivial to link someone to the right docs directly when we're giving support